src/test/java/com/epam/tc/hw2/ex1/Ex1.java

```diff
@@ -0,0 +1,174 @@
+                .as("Incorrect METALS & COLORS button name").isEqualTo("METALS & COLORS");
+
+        //6. Assert that there are 4 images on the Index Page and they are displayed
+        List<WebElement> icons = driver.findElements(By.cssSelector(".benefit-icon"));
+        softly.assertThat(icons.size()).as("Number of images are " + icons.size() + " not 4").isEqualTo(4);
+        softly.assertThat(icons).as("One or more image not displayed").isNotEmpty();
+        //other way: assertTrue(isImageDisplayed);
+
+        //7. Assert that there are 4 texts on the Index Page under icons and they have proper text
+        List<WebElement> iconTexts = driver.findElements(By.cssSelector(".benefit-txt"));
+        softly.assertThat(iconTexts.size())
+                .as("Number of texts are " + iconTexts.size() + " not 4").isEqualTo(4);
+        softly.assertThat(iconTexts.get(0).getText())
+                .as("To include good...text differs")
+                .isEqualTo("To include good practices\nand ideas from successful\nEPAM project");
+        softly.assertThat(iconTexts.get(1).getText())
+                .as("To be flexible...text differs").isEqualTo("To be flexible and\ncustomizable");
+        softly.assertThat(iconTexts.get(2).getText())
+                .as("To be multiplatform").isEqualTo("To be multiplatform");
+        softly.assertThat(iconTexts.get(3).getText())
+                .as("Already have good base...text differs")
+                .isEqualTo("Already have good base\n"
+                        + "(about 20 internal and\n"
+                        + "some external projects),\nwish to get more…");
+
+        //8. Assert that there is the iframe with “Frame Button” exist
+        int size = driver.findElements(By.tagName("iframe")).size();
+        System.out.println(size);
+        int buttonInCurrentFrame = 0; //count buttons “Frame Button” in current frame
+        int totalButtonsInFrames = 0; //counter of frames with button “Frame Button”
+        for (int i = 0; i < size; i++) {
+
+            //9. Switch to the iframe and check that there is “Frame Button” in the iframe
+            driver.switchTo().frame(i);
+            buttonInCurrentFrame = driver.findElements(By.cssSelector("input#frame-button")).size();
+
+            if (buttonInCurrentFrame > 0) {
+                totalButtonsInFrames = 0 + buttonInCurrentFrame;
+                WebElement frameButton = driver
+                        .findElement(By.cssSelector("input#frame-button"));
+                softly.assertThat(frameButton.getAttribute("value"))
+                        .as("Frame Button text differs")
+                        .isEqualTo("Frame Button");
+            }
+
+            System.out.println(buttonInCurrentFrame);
+
+            //10. Switch to original window back
+            driver.switchTo().defaultContent();
+        }
+        softly.assertThat(totalButtonsInFrames).as("Button not found").isNotZero();
```

Comment:
The idea was to find frame with id=frame(`#frame`). Switch into it and check the button.

But you choose to check all frames on a page for `"Frame Button"`. But your solution is ok

Reply:
But `totalButtonsInFrames` is not total buttons amount in all frames. It's number of buttons in the first frame that contains these buttons. You don't have any SUM to get total buttons amount. Rename or fix it somehow

Reply:
Counter fixed